specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md

```diff
@@ -110,13 +110,241 @@ input-file:
 - track1/preview/2019-05-06-preview/searchindex.json
 ```
 
+### Tag: track2-package-2019-05
+
+These settings apply only when `--tag=track2-package-2019-05` is specified on the command line.
+It is meant to be used for generating Track 2 Java SDK only!
+
+``` yaml $(tag) == 'track2-package-2019-05'
+input-file:
+- track2/stable/2019-05-06/searchindex.json
+```
+
+### Tag: track2-package-2019-05-preview
+
+These settings apply only when `--tag=track2-package-2019-05-preview` is specified on the command line.
+It is meant to be used for generating Track 2 Java SDK only!
+
+``` yaml $(tag) == 'track2-package-2019-05-preview'
+opt-in-extensible-enums: true
+input-file:
+- track2/preview/2019-05-06-preview/searchindex.json
+```
+
 ---
 # Code Generation
 
 !!! READ THIS !!!
-This swagger is not yet ready for languages other than C#.
+This swagger is ready for C# and Java
 !!! READ THIS !!!
 
+## Java 
+
+These settings apply only when `--java` is specified on the command line.
+Please also specify `--java-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(java)
+title: SearchIndexRestClient
+output-folder: $(java-sdks-folder)/search/Microsoft.Azure.Search.Data/src/Generated
+namespace: com.azure.search
+java: true
+sync-methods: none
+add-context-parameter: true
+generate-client-interfaces: false
+license-header: |-
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT License.
+  Code generated by Microsoft (R) AutoRest Code Generator.
+  Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+require: ../autorest-custom-directives.md
+
+vararg-properties: >-
+  AutocompleteOptions.searchFields,
+  SearchOptions.facets, SearchOptions.highlightFields, SearchOptions.orderBy, SearchOptions.scoringParameters, SearchOptions.searchFields, SearchOptions.select,
+  SuggestOptions.orderBy, SuggestOptions.searchFields, SuggestOptions.select
+```
+
+### Tweak generated code
+
+Use the below directives sparingly. Every modification we make in here, we potentially have to replicate for every target language.
+
+``` yaml $(java)
+directive:
+# Rename custom properties
+- rename-custom-property:
+    path: "$.paths.*.*.parameters[*]['x-ms-parameter-grouping']"
+    from: AutocompleteParameters
+    to: AutocompleteOptions
```

Comment:
Now that we have a separate spec for Track 2, could this be accomplished by simply renaming the type in the Swagger instead of having a directive? That would be preferable, provided the naming conventions apply across languages.

Reply:
@brjohnstmsft - who is the person to ask for changes in the Swagger? and who can validate and approve that the naming conventions apply across languages?

Reply:
Anyone is free to submit a PR to change the Swagger -- you or someone else on your team can, for example. I'd just ask that you add me as a reviewer to that PR.

Regarding the naming conventions, the Azure API Guidelines are published [here](https://azure.github.io/azure-sdk/general_design.html). If there are ambiguities in the written guidelines, @adrianhall should be able to address them.

Reply:
I have opened a User Story for changes in the Swagger:
https://dev.azure.com/csee2e/Azure%20Search%20SDK/_workitems/edit/1716

Reply:
Once https://github.com/Azure/azure-rest-api-specs/pull/7978 is approved, this configuration should update and remove the usage of rename-custom-property